app/assets/javascripts/discourse/app/components/modal/history.js

```diff
@@ -110,8 +112,9 @@ export default class History extends Component {
           historyOneboxes: ["header", "article", "div[style]"],
         },
       };
-      sanitizeAsync(html, opts).then((result) => (html = result));
-      return html;
+      sanitizeAsync(html, opts).then((result) => {
+        return result;
```

Comment:
Nothing is listening to the result of this promise, so this `return` statement is just going to disappear into the ether.

Async getters aren't really a thing... so I think the only strategy here is something like

```
@tracked bodyDiff;

@action
async calculateBodyDiff(){
  let html = ...;
  if(){
    ...
  } else {
    html = await sanitizeAsync(html, opts);
  }
  this.bodyDiff = html;
}
```

and then trigger `calculateBodyDiff` using `{{did-insert}}` and `{{did-update}}`